### DISCUSSION TITLE
Enum discovery misses nested enums

### DISCUSSION THREAD

**Discussion by ccjernigan** [LemonAppDev/konsist#419] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.11.0

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm writing this rule:

```
    @Test 
    fun `require explicit serialName for enum constants`() {
        Konsist.scopeFromProject()
            .classes()
            .withSomeAnnotationsOf(kotlinx.serialization.Serializable::class)
            .withSomeModifiers(KoModifier.ENUM)
            .enumConstants
            .assert {
                it.hasAnnotationsOf(kotlinx.serialization.SerialName::class)
            }
    }
```

To test against this code:

```
@Serializable
sealed class PasswordAlgorithm {
    @Serializable
    @SerialName("pbkdf2")
    data class PBKDF2(
        @SerialName("a")
        val algorithm: HashingAlgorithm,
        @SerialName("i")
        val iterations: Int
    ) : PasswordAlgorithm() {

        @Serializable
        enum class HashingAlgorithm {
            SHA256
        }
    }
}
```

### Actual behaviour

Test passes.

The enum doesn't even get picked up (e.g. if I do a print()).

If I un-nest the enum and move it to the top level, then it does get picked up.

### Expected behaviour

The test case should fail because the SHA256 enum constant is not annotated.

### Relevant code snippet.

```kotlin
Copy/paste the code snippet here, between the starting and ending backticks
```

</details>

**Comment by igorwojda** [comment, marked as answer]
By default nested classes (and local functions as not listed).
For now please add `.classes(includeNested = true)`.

In next release `true` will be a default value for `includeNested` (and `includeLocal`)
